pkg/frontend/querymiddleware/codec_protobuf.go

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package querymiddleware
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/prometheus/common/model"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+)
+
+type protobufFormat struct{}
+
+func (f protobufFormat) Name() string {
+	return formatProtobuf
+}
+
+func (f protobufFormat) EncodeResponse(resp *PrometheusResponse) ([]byte, error) {
+	panic("not yet implemented")
```

Comment:
Note to reviewers: this method is never called, so this panic is safe. (We only ever use the JSON format in `prometheusCodec.EncodeResponse()`.)